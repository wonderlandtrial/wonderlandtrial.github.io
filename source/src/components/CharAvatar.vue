<template>
<v-avatar
  :size="84"
  @mouseenter="$emit('mouseenter', '')">
  <img :src="value.potrait" :alt="value.name" />
</v-avatar>
</template>

<script>
export default {
  name: 'CharAvatar',
  props: ['value']
}
</script>
