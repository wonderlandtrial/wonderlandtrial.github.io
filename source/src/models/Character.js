class Character {
  constructor(
    name = '',
    position = 'monster',
    skins = []
  ) {
    this.name = name
    this.position = position
    this.skins = skins

    if (!name) {
      if (position === 'gold') {
        this.potrait = '/static/assets/potraits/gold_potrait.png'
      } else if (position === 'black') {
        this.potrait = '/static/assets/potraits/black_potrait.png'
      } else if (position === 'white') {
        this.potrait = '/static/assets/potraits/white_potrait.png'
      } else {
        this.potrait = '/static/assets/potraits/logo.jpg'
      }
    } else {
      this.potrait = `/static/assets/potraits/${this.getPotraitPath()}_potrait.png`
    }
  }

  getPotraitPath() {
    return this.name.toLocaleLowerCase().split(' ').join('_')
  }
}

export default Character
