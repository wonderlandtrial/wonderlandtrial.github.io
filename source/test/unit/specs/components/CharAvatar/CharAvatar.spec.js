import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'
import 'babel-polyfill'

import CharAvatar from '@/components/CharAvatar'
import Character from '@/models/Character'

const localVue = createLocalVue()
localVue.use(Vuetify)

describe('CharAvatar', () => {
  it('rendered result should match snapshot', () => {
    let char = new Character('Fatima', 'black')

    const wrapper = mount(CharAvatar, {
      propsData: {
        value: char
      },
      localVue
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
