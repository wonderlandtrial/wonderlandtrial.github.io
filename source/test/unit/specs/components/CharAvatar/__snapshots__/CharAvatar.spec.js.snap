// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharAvatar rendered result should match snapshot 1`] = `

<div class="avatar"
     style="height: 84px; width: 84px;"
>
  <img src="/static/assets/potraits/fatima_potrait.png"
       alt="Fatima"
  >
</div>

`;
